src/main/java/duke/CommandHandler.java

```diff
@@ -0,0 +1,276 @@
+package duke;
```

Comment:
`CommandHandler.java`

There is a lack of SRP for this class. You could isolate methods that do not really fit well inside this class. For example, task-related, UI-related, and pure utility functions could be extracted out to dedicated classes.